### DISCUSSION TITLE
PstFrom setup with external .pst-file

### DISCUSSION THREAD

**Discussion by d-strom** [pypest/pyemu#495] [Q&A]
Hi!

Suppose I have an external .pst-file and dont want to go through the hassle of importing csv or text files for my observations and other parameters (since they are already set up in there). Is there any cleaver ways to extract this information from my "old" pest file into a PstFrom class?

For instance in a model I'm working with I have areas of constant conductivity represented by different soil material in my base model where the upper and lower bounds differ depending on the type of soil. In my current .pst-file this is already set up. However, should I have to set it up again for the PstFrom class I'm not sure how to separate the zones from each other.

Sincerely

David

**Comment by briochh** [comment]
Hi @DStrom1987,
Sorry for being slow to this. Unfortunately there currently isn't a totally straightforward way from getting from an existing control file to a PstFrom class. A followup question from me would be what functions of the PstFrom class are you needing? Perhaps there is another way to get into it. @wkitlasten has been looking into abstracting the `PstFrom.draw()`method to make it accessible from outside PstFrom (see #514). Would this sort of thing be helpful?
